### PR TITLE
Fix 4535 improved and uniformed feedback for resources

### PR DESCRIPTION
**Pull request by MV88**
## Description
Fixes some problems on edit properties modal for gestore resources (maps, dashboards, stories)

## Issues
 - #4535

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [x] Feature

**What is the current behavior?** (You can also link to an open issue here)
see issue

**What is the new behavior?**
- Error messages has ben generalized cor maps, dashboards and stories
- a successful notification has been added for the three above
- in homepage if you save and use an existing name, the error will appear, if you close and reopen the same card the error is gone
- save and save as in map no longer keeps the error message

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes, and I documented them in migration notes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

@offtherailz please note that some of the code has been done again on the map component actually being used

### CODE REVIEW

File: web/client/plugins/Save.jsx [hunk 1]
```diff
@@ -26,7 +27,8 @@ export default createPlugin('Save', {
             mapInfoSelector,
             (show, resource) => ({show, resource})),
         {
-            onClose: toggleControl.bind(null, 'mapSave', false)
+            onClose: toggleControl.bind(null, 'mapSave', false),
+            onMapSaveError: mapSaved
```
Comment (offtherailz) [inline web/client/plugins/Save.jsx]: This looks a little strange to me? mapSaved bind to mapSaveError?

File: web/client/actions/dashboards.js [hunk 2]
```diff
@@ -14,7 +14,11 @@ const DASHBOARD_DELETED = "DASHBOARDS:DASHBOARD_DELETED";
 const RELOAD = "DASHBOARDS:RELOAD_DASHBOARDS";
 const LOADING = "DASHBOARDS:LOADING";
 
+const { success: successNotification } = require('./notifications');
+
+
 module.exports = {
+    showSuccessNotification: () => successNotification({ title: "success", message: "resources.successSaved" }),
```
Comment (offtherailz) [inline web/client/actions/dashboards.js]: This is replicated 3 times in 3 files and then again in the files. 
This is not necessary. There is a little of duplication for these tools that have to be put in common. Please include these declarations in the plugins.

File: web/client/plugins/Save.jsx (hunk 1 above)
Comment (MV88) [reply web/client/plugins/Save.jsx]: yes, it is used to [reset errors](https://github.com/geosolutions-it/MapStore2/blob/master/web/client/reducers/config.js#L118) when edit properties modal is closed..

going to change the name to
```suggestion
            onResetMapSaveError: mapSaved
```

Comment (offtherailz) [reply web/client/plugins/Save.jsx]: Create a specific action for it, otherwise is confusing

Comment (offtherailz) [reply web/client/plugins/Save.jsx]: Or you create a RESET_ERRORS action or you explain that you are using the mapSaved action to reset error. I'm not sure if you pass the map or not. Please explain and find out a better name.

onResetMapSaveError doesn't make to much sense. what it means? "on Reset, MapSaveError"/ on reset map, save error"...
handlers name with "on" means that they are called in sequence of an event. 
onReset means that is called when button "Reset" is clicked, or a "reset" action is performed by the user, anyway,
onPan means the handler is called when the map pans.

Instead, if the property is instead a function that have to be called to have some effect, you can call it "reset". 
"resetError" will reset the error.

Comment (MV88) [reply web/client/plugins/Save.jsx]: i could create another constant in config.js action file that does the same thing as MAP_SAVED
i wanted to reuse it, gonna create it anyway and change the name of the action passed to the plugin to resetErrors

Comment (MV88) [reply web/client/plugins/Save.jsx]: *correction gonna call it **clearErrors**

**Comment by MV88** [comment]
closing this in favor of https://github.com/geosolutions-it/MapStore2/pull/4592

**Pull request closed without merging**